pyiron_contrib/atomistics/atomicrex/structure_list.py

```diff
@@ -517,6 +526,83 @@ def prepare_plotting(self, final_values=False):
             * self._structures.length
         )
 
+    #### PotentialFit methods
+    def add_training_data(self, container: TrainingContainer) -> None:
+        storage = container._container
+        atomic_energy_storage = FlattenedARScalarProperty(
+            num_chunks=storage.num_chunks, num_elements=storage.num_elements
+        )
+        atomic_energy_storage.num_chunks = storage.num_chunks
+        atomic_energy_storage.num_elements = storage.num_elements
+        atomic_energy_storage._per_chunk_arrays["fit"][0 : storage.num_chunks] = True
+        atomic_energy_storage._per_chunk_arrays["tolerance"][
+            0 : storage.num_chunks
+        ] = 0.001
```

Comment:
I guess you are breaking API here, because you want to set the same data more efficiently?  In that case, we should probably move this use case directly to `FlattenedStorage`, maybe by letting `set_array` accept `slice`.

I'm ok with leaving this for the workshop though, if you make an issue to remind us in base.

Reply:
This is necessary because it overwrites the standard values. In the way I normally add structures this doesn't matter, but typically energies are weighted higher than forces so I wanted a more reasonable choice for these values.

Reply:
Now I get it :)